xir/parser.py

```diff
@@ -260,16 +260,30 @@ def obs_stmt(self, args):
             ObservableStmt: object containing statement data
         """
         pref = simplify_math(args[0])
-        terms = args[1]
-        return ObservableStmt(pref, terms, use_floats=self.use_floats)
+        factors = args[1]
+        return ObservableStmt(
+            pref, [ObservableFactor(*factor) for factor in factors], use_floats=self.use_floats
+        )
 
     def obs_group(self, args):
         """Group of observables used to define an observable statement.
 
         Returns:
             list[tuple]: each observable with corresponding wires as tuples
         """
-        return [(args[i], args[i + 1]) for i in range(0, len(args) - 1, 2)]
+        obs_group_list = []
```

Comment:
We should update the docstring to specify the actual items returned in each tuple. Otherwise, this function is basically unusable without reading through its implementation.